vpr/src/place/simpleRL_move_generator.cpp

```diff
@@ -0,0 +1,408 @@
+#include "simpleRL_move_generator.h"
+#include "globals.h"
+#include <algorithm>
+#include <numeric>
+
+#include "vtr_random.h"
+
+float my_exp(float);
+
+//EpsilonGreedyAgent member functions
+EpsilonGreedyAgent::EpsilonGreedyAgent(size_t k, float epsilon) {
+    //f_ = vtr::fopen("agent_info.txt", "w");
+    //VTR_LOG("!!!!%f\n",epsilon);
+    set_epsilon(epsilon);
+    set_k(k);
+    set_epsilon_action_prob();
+
+    //string log_file = "agent_info.txt";
+    //mode = "w";
+    //f_ = fopen(log_file, mode);
+}
+
+EpsilonGreedyAgent::~EpsilonGreedyAgent() {
+    if (f_) vtr::fclose(f_);
+}
+
+void EpsilonGreedyAgent::set_step(float gamma, int move_lim) {
+    //VTR_LOG("Setting egreedy step: %g\n", exp_alpha_);
+    if (gamma < 0) {
+        exp_alpha_ = -1; //Use sample average
+    } else {
+        //
+        // For an exponentially wieghted average the fraction of total weight applied to
+        // to moves which occured > K moves ago is:
+        //
+        //      gamma = (1 - alpha)^K
+        //
+        // If we treat K as the number of moves per temperature (move_lim) then gamma
+        // is the fraction of weight applied to moves which occured > move_lim moves ago,
+        // and given a target gamma we can explicitly calcualte the alpha step-size
+        // required by the agent:
+        //
+        //     alpha = 1 - e^(log(gamma) / K)
+        //
+        float alpha = 1 - std::exp(std::log(gamma) / move_lim);
+        //VTR_LOG("K-armed bandit alpha: %g\n", alpha);
+        exp_alpha_ = alpha;
+    }
+}
+
+void EpsilonGreedyAgent::process_outcome(double reward, int reward_num) {
+    ++n_[last_action_];
+    if (reward_num == 1 || reward_num == 4) {
```

Comment:
What is `reward_num`? This should probably be an `enum`.

Reply:
changed it to be a string that has a meaningful option names